snapcraft/_options.py

```diff
@@ -14,16 +14,120 @@
+        'kernel': 'arm64',
+        'deb': 'arm64',
+        'cross-compiler-prefix': 'aarch64-linux-gnu-',
+        'cross-build-packages': ['gcc-aarch64-linux-gnu'],
+        'triplet': 'aarch64-linux-gnu',
+    },
+    'i686': {
+        'kernel': 'x86',
+        'deb': 'i386',
+        'triplet': 'i386-linux-gnu',
+    },
+    'ppc64le': {
+        'kernel': 'powerpc',
+        'deb': 'ppc64el',
+        'cross-compiler-prefix': 'gcc-powerpc64-linux-gnu-',
+        'cross-build-packages': ['gcc-powerpc64-linux-gnu'],
+        'triplet': 'powerpc64le-linux-gnu',
+    },
+    'x86_64': {
+        'kernel': 'x86',
+        'deb': 'amd64',
+        'triplet': 'x86_64-linux-gnu',
+    }
+}
+
 
 class ProjectOptions:
 
     @property
     def use_geoip(self):
         return self.__use_geoip
 
-    @use_geoip.setter
-    def use_geoip(self, setting):
-        self.__use_geoip = setting
+    @property
+    def parallel_builds(self):
+        return self.__parallel_builds
+
+    @property
+    def parallel_build_count(self):
+        build_count = 1
+        if self.__parallel_builds:
+            try:
+                build_count = multiprocessing.cpu_count()
+            except NotImplementedError:
+                logger.warning(
+                    'Unable to determine CPU count; disabling parallel builds')
+
+        return build_count
+
+    @property
+    def is_cross_compiling(self):
+        return self.__target_machine != self.__host_machine
+
+    @property
+    def cross_compiler_prefix(self):
+        try:
+            return self.__machine_info['cross-compiler-prefix']
+        except KeyError:
+            raise EnvironmentError(
+                'Cross compilation not support for target arch {!}'.format(
+                    self.__machine_target))
+
+    @property
+    def additional_build_packages(self):
+        return self.__machine_info.get('cross-build-packages', [])
+
+    @property
+    def arch_triplet(self):
+        return self.__machine_info['triplet']
+
+    @property
+    def deb_arch(self):
+        return self.__machine_info['deb']
+
+    @property
+    def kernel_arch(self):
+        return self.__machine_info['kernel']
+
+    def __init__(self, use_geoip=False, parallel_builds=True,
+                 target_deb_arch=None):
+        self.__use_geoip = use_geoip
+        self.__parallel_builds = parallel_builds
+        self._set_machine(target_deb_arch)
+
+    def _set_machine(self, target_deb_arch):
+        self.__host_machine = platform.machine()
+        if not target_deb_arch:
+            self.__target_machine = self.__host_machine
+        else:
+            self.__target_machine = _find_machine(target_deb_arch)
+        self.__machine_info = _ARCH_TRANSLATIONS[self.__target_machine]
+
+
+def _find_machine(deb_arch):
+    for machine in _ARCH_TRANSLATIONS:
+        if _ARCH_TRANSLATIONS[machine].get('deb', '') == deb_arch:
+            logger.info('Setting target machine to {!r}'.format(machine))
```

Comment:
This function isn't actually setting anything-- this `logger.info()` probably belongs in `_set_machine()`, no?